pkg/rpc/request/txBuilder.go

```diff
@@ -64,10 +64,15 @@ func ExpandArrayIntoScript(script *io.BinWriter, slice []Param) error {
 			emit.Bytes(script, key.Bytes())
 		case smartcontract.IntegerType:
 			val, err := fp.Value.GetInt()
+			if err == nil {
+				emit.Int(script, int64(val))
+				continue
+			}
+			bi, err := fp.Value.GetBigInt()
```

Comment:
Why bother with `Value.GetInt/emit.Int` then?

Reply:
In some RPC small integers are expected (paging, block indices), using `big.Int` there has some costs (allocation for the integer itself + additional allocations for any simple operations). There are only a few such places though.

Reply:
And here we're specifically creating a script, I'm not suggesting remove `GetInt` entirely, it's just here it makes little sense.

Reply:
Fixed.